src/main/java/tools/jackson/databind/DeserializationContext.java

```diff
@@ -1354,6 +1354,45 @@ public Object handleWeirdStringValue(Class<?> targetClass, String value,
         throw weirdStringException(value, targetClass, msg);
     }
 
+    /**
+     * Method that deserializers should call if they encounter a null value and
+     * target value type is a Primitive type.
+     *
+     * Default implementation will try to call {@link DeserializationContext#reportInputMismatch(Class, String, Object...)},
+     * which by default would throw {@link MismatchedInputException}
+     *
+     * @param targetClass Primitive type into which incoming {@code null} value should be converted to
+     * @param deser Type of {@link ValueDeserializer} calling this method
+     * @param msg Error message template caller wants to use if exception is to be thrown
+     *
+     * @throws JacksonException To indicate unrecoverable problem, usually based on <code>msg</code>
+     */
+    public Object handleNullForPrimitives(Class<?> targetClass,
+                                          ValueDeserializer<?> deser, String msg)
+            throws JacksonException
+    {
+        // but if not handled, just throw exception
+        LinkedNode<DeserializationProblemHandler> h = _config.getProblemHandlers();
+        while (h != null) {
+            // Can bail out if it's handled
+            Object instance = h.value().handleNullForPrimitives(this, targetClass, deser, _parser, msg);
+            if (instance != DeserializationProblemHandler.NOT_HANDLED) {
+                // Sanity check for broken handlers, otherwise nasty to debug:
+                if (_isCompatible(targetClass, instance)) {
+                    return instance;
+                }
+                // In case our problem handler providing incompatible value,
+                throw new InvalidFormatException(_parser,
+                        String.format("Cannot deserialize value of type %s from type %s",
+                                targetClass, ClassUtil.getClassDescription(instance)),
+                                  instance, targetClass
```

Comment:
Instead of plain `targetClass`, use method from `ClassUtil`, I think `nameOf(Class<?>)`.
Handles `null`s and surrounds with backtick (plus I think deals with array class types).